content/blog/ggplot2-3-4-0/index.Rmd

```diff
@@ -0,0 +1,220 @@
+---
+output: hugodown::hugo_document
+
+slug: ggplot2-3-4-0
+title: ggplot2 3.4.0
+date: 2022-11-01
+author: Thomas Lin Pedersen
+description: >
+    ggplot2 3.4.0 is now on CRAN. Read all about the (mostly internal) changes
+    that makes up this release.
+
+photo:
+  url: https://data-imaginist.com/art
+  author: Thomas Lin Pedersen
+
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [package] 
+tags: [ggplot2, graphics]
+---
+
+```{=html}
+<!--
+TODO:
+* [ ] Look over / edit the post's title in the yaml
+* [ ] Edit (or delete) the description; note this appears in the Twitter card
+* [ ] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [ ] Find photo & update yaml metadata
+* [ ] Create `thumbnail-sq.jpg`; height and width should be equal
+* [ ] Create `thumbnail-wd.jpg`; width should be >5x height
+* [ ] `hugodown::use_tidy_thumbnails()`
+* [ ] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+```
+We're so happy to announce the release of [ggplot2](https://ggplot2.tidyverse.org) 3.4.0 on CRAN. ggplot2 is a system for declaratively creating graphics, based on The Grammar of Graphics. You provide the data, tell ggplot2 how to map variables to aesthetics, what graphical primitives to use, and it takes care of the details. The new version can be installed from CRAN using `install.packages("ggplot2")`.
+
+This release is not full of exciting new features. Instead we have focused on the internals, tightening up of the API, and improving the messaging, especially when it comes to errors and warnings. While the release also contains a few new features these other aspects are the stars of this release.
+
+You can see a full list of changes in the [release notes](https://ggplot2.tidyverse.org/news/index.html)
+
+```{r setup, message=FALSE}
+library(ggplot2)
+library(patchwork)
+library(dplyr)
+```
+
+## Hello `linewidth`
+
+Arguably the biggest user.visible change in this release is the introduction of a new fundamental aesthetic. `linewidth` will from this point on take over sizing of the width of lines---something that was earlier handled by `size`. The reason for this change is that prior to this release `size` was used for two related, but different, properties: the size of points (and glyphs) and the width of lines. Since one is area based and one is length based they fundamentally needs different scaling and the default size scale has always catered to area sizing, using a square root transform. Further, this conflation has also resulted in composite geoms like `geom_pointrange()` where it was difficult to control the line width and point size separately.
```

Comment:
```suggestion
Arguably the biggest user.visible change in this release is the introduction of a new fundamental aesthetic. From this release on, `linewidth` will take over sizing of the width of lines---something that was earlier handled by `size`. The reason for this change is that prior to this release `size` was used for two related, but different, properties: the size of points (and glyphs) and the width of lines. Since one is area based and one is length based they fundamentally needs different scaling and the default size scale has always catered to area sizing, using a square root transform. This conflation has also made it hard for composite geoms like `geom_pointrange()` to control the line width and point size separately.
```